redaxo/src/core/lib/Fragment/Component/Alert.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Redaxo\Core\Fragment\Component;
+
+use Redaxo\Core\Fragment\Slot;
+use rex_fragment;
+
+class Alert extends rex_fragment
```

Comment:
Wollen wir mit Fragment-spezifischen Klassen arbeiten?
Also ein Fragment besteht dann aus zwei Dateien, aus der Klasse und dem eigentlichen Fragment (HTML-Template).

Dadurch bekämen wir eine bessere Typsicherheit. Sowohl bei der Aufrufstelle, als auch im eigentlichen Fragment wären dann die verfügbaren Properties mit den jeweiligen Typen bekannt.

Reply:
in meinen augen ist der haupt-schwachpunkt von fragmenten aktuell die schlechte statische analysierbarkeit.

in meinen augen ist das der haupt treiber für diesen ansatz hier?

Reply:
Ja, statische Analyse, und aber auch autocomplete in phpstorm.

Reply:
Idee ist aber auch, dass man in der Fragment-Klasse ggf. auch mal noch spezifische Methoden für das Fragment notieren kann, um die Logik im Fragment-Template reduzieren zu können.